pkg/route/hostassignment/externalcertificate_test.go

```diff
@@ -0,0 +1,209 @@
+package hostassignment
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	routev1 "github.com/openshift/api/route/v1"
+	routecommon "github.com/openshift/library-go/pkg/route"
+)
+
+func TestValidateHostExternalCertificate(t *testing.T) {
+
+	type args struct {
+		ctx   context.Context
+		new   *routev1.Route
+		older *routev1.Route
+		allow bool
+		opts  routecommon.RouteValidationOptions
+	}
+	tests := []struct {
+		name string
+		args args
+		want field.ErrorList
+	}{
+		{
+			name: "Updating new route and old route with nil TLS",
+			args: args{
+				ctx: request.WithUser(context.Background(), &user.DefaultInfo{Name: "user1"}),
+				new: &routev1.Route{
+					Spec: routev1.RouteSpec{},
+				},
+				older: &routev1.Route{
+					Spec: routev1.RouteSpec{},
+				},
+				allow: false,
+				opts:  routecommon.RouteValidationOptions{AllowExternalCertificates: true},
+			},
+			want: field.ErrorList{},
```

Comment:
`field.ErrorList{}` is equivalent to `nil`? Hm, how the error check should look like on the caller side?
```go
if errs := ValidateHostExternalCertificate(); len(errs) != 0 {}
```
Also, `nil` can be returned too. I think just `nil` as "no error" would be the most idiomatic.